### PR TITLE
Remove unwind priority.

### LINKED ISSUE #567: Remove `unwind_priority` in `Pipeline`
### Describe the feature

We *should* be able to unwind in direct reverse order of the order we sync in, so this is added complexity that we are not going to use.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TechieBoy**
This removes the unwind_priority field from QueuedStage. Unwinds now only happen in direct reverse order of insertion into the pipeline.
Closes #567

### CODE REVIEW

File: crates/stages/src/pipeline.rs [hunk 1]
```diff
@@ -68,11 +68,8 @@ use state::*;
 /// # Unwinding
 ///
 /// In case of a validation error (as determined by the consensus engine) in one of the stages, the
-/// pipeline will unwind the stages according to their unwind priority. It is also possible to
+/// pipeline will unwind the stages in reverse order of insertion. It is also possible to
```
Comment (rkrasiuk) [inline crates/stages/src/pipeline.rs:71]:
```suggestion
/// pipeline will unwind the stages in reverse order of execution. It is also possible to
```

File: crates/stages/src/pipeline.rs [hunk 2]
```diff
@@ -108,19 +105,11 @@ impl<DB: Database> Pipeline<DB> {
     /// # Unwinding
     ///
     /// The unwind priority is set to 0.
```
Comment (rkrasiuk) [inline crates/stages/src/pipeline.rs:107]: remove this comment

File: crates/stages/src/pipeline.rs [hunk 3]
```diff
@@ -217,7 +206,6 @@ impl<DB: Database> Pipeline<DB> {
         // Sort stages by unwind priority
         let mut unwind_pipeline = {
             let mut stages: Vec<_> = self.stages.iter_mut().enumerate().collect();
-            stages.sort_by(|a, b| a.1.unwind_priority.cmp(&b.1.unwind_priority));
             stages.reverse();
             stages
         };
```
Comment (rkrasiuk) [inline crates/stages/src/pipeline.rs:211]: unwind pipeline can be kept as an iterator `let unwind_pipeline = self.stages.iter_mut().rev();``

**Review by rkrasiuk — CHANGES_REQUESTED** [review]
Nice, thanks! smol nits left

**Review by gakonst — APPROVED** [review]
lgtm pending roman's nits

**Comment by TechieBoy** [comment]
Resolved all the nits, should be good to merge now!

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**